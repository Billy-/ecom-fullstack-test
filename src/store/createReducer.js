const createReducer = (initialState, actionHandlers) => {
    return function reducer (state = initialState, action) {
        const handler = actionHandlers[action.type];
  
        return handler ? handler(state, action) : state;
    };
};

export default createReducer;
