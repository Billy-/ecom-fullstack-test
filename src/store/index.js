export { default } from './store';

import './modules';
