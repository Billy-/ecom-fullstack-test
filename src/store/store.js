import { createStore } from 'redux';

const getComposeEnhancers = () => {
    /* Hack for SSR */
    if (!global.window) global.window = {};
    // eslint-disable-next-line no-underscore-dangle
    return window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
};

// create Redux store
const store = createStore(
    // no reducers at init time, they are injected async
    x => x,
    getComposeEnhancers(),
);

store.asyncReducers = {};

export default store;

