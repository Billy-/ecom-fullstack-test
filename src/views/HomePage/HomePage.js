import ProductList from  '../../components/ProductList';
import withProducts from '../../HOCs/withProducts';

const HomePage = withProducts(ProductList);

export default HomePage;
