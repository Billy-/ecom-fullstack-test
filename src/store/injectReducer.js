import { combineReducers } from 'redux';

import store from './store';

const injectReducer = (key, reducer) => {
    if (Object.hasOwnProperty.call(store.asyncReducers, key)) return;

    store.asyncReducers[key] = reducer; // eslint-disable-line no-param-reassign

    store.replaceReducer(combineReducers(store.asyncReducers));
};

export default injectReducer;
