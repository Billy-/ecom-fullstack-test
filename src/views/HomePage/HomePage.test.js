import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from '../../store';

import HomePage from './HomePage';

const props = {};

describe('HomePage', () => {
    it('renders without throwing', () => {
        renderer.create(
            <Provider store={store}>
                <HomePage {...props} />
            </Provider>);
    });

    it('matches snapshot', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <HomePage {...props} />
                </Provider>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
