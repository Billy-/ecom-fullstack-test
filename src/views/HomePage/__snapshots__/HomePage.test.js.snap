// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage matches snapshot 1`] = `
<div>
  <h1>
    Loading...
  </h1>
</div>
`;
